### PR TITLE
Add check_bucket_exists function

### PR DESCRIPTION
**Pull request by amork**
Add GCP version of the check_bucket_exists function.
The function is to be initially used by the HCA DSS checkout service.

Add a new unit test - positive and negative test of a bucket existence

### CODE REVIEW

**Comment by ttung** [comment]
Based on the test failures, I think the permissions need to be tweaked on GS bucket `czi-cloud-blobstore-fixtures`

File: tests/test_gsblobstore.py [hunk 1]
```diff
@@ -39,5 +39,15 @@ def test_get_checksum(self):
                 self.test_fixtures_bucket,
                 "test_good_source_data_DOES_NOT_EXIST")
 
+    def test_check_bucket_exists(self):
```
Comment (ttung) [inline tests/test_gsblobstore.py]: The test in tests/test_s3blobstore.py should be moved to blobstore_common_tests.  Then it'll be run against both stores.

**Review by ttung — CHANGES_REQUESTED** [review]

File: tests/test_gsblobstore.py (hunk 1 above)
Comment (amork) [reply tests/test_gsblobstore.py]: done
